src/transformers/models/t5/modeling_flax_t5.py

```diff
@@ -155,6 +155,44 @@ def __call__(self, hidden_states, deterministic):
         return hidden_states
 
 
+class FlaxT5DenseGatedSiLUDense(nn.Module):
```

Comment:
Thanks a lot for the PR @DanielHesslow !

Super nice that you look into it. I think the problem here is not really that we need a new `GatedSiluDense` module class, but rather that the naming of `GatedGeluDense` is badly chosen and that we should make the activation function more configurable with the config. 

Could you maybe change your PR slighly to do the following:

1. add a parameter `dense_act_fn` to `configuration_t5.py` that will default to `None`. Then if `config.feed_forward_proj == "relu"` will **and** `dense_act_fn == None`, we can set it to `"relu"` otherwise we'll set it to `"gelu_new"`. This logic should happen in `configuration_t5.py` and will ensure 100% backward comp
2. Rename all `...DenseGated{Relu,Gelu}Dense` modules to `...DenseGatedActDense` and `...DenseActDense` 
3. Replace all `ACT2FN["gelu_new"]` , `ACT2FN["relu"]` with `ACT2FN[config.dense_act_fn]`

Does this make sense? 

@patil-suraj what do you think?

Reply:
Yeah not reusing the code is ugly, definitely agree. 
Don't quite see how your solution would work tho, wouldn't you need some way to switch between gated act vs non-gated act?

I guess smth like the following would work fine tho:
```
// conf_t5.py
def __init__(
    ...
    is_gated_act=False
):

    if feed_forward_proj == "gated-gelu":
         self.dense_act_fn = "gelu_new"
         self.is_gated_act = True
    else:
         self.dense_act_fn = self.feed_forward_proj
         self.is_gated_act = is_gated_act
    
    # keep this around to ensure complete back compat.
    self.feed_forward_proj = feed_forward_proj
```